datafusion-examples/examples/simple_udaf.rs

```diff
@@ -124,6 +124,34 @@ impl Accumulator for GeometricMean {
     // Optimization hint: this trait also supports `update_batch` and `merge_batch`,
     // that can be used to perform these operations on arrays instead of single values.
     // By default, these methods call `update` and `merge` row by row
```

Comment:
I suggest we update this example (I can do it as a follow on PR) to not use ScalarValue at all

```suggestion
    // DataFusion calls this function to update the accumulator's state for a batch 
    // of inputs rows. In this case the product is updated with values from the first column
    // and the count is updated based on the row count
```